### PR TITLE
docs(layout containers) add flexbox layout content

### PR DESCRIPTION
**Pull request by manoldonev**
- Updated code snippets
- Added FlexboxLayout content

I think https://github.com/NativeScript/nativescript-angular/pull/1124 has to be merged first so the ng code snippets can be extracted correctly.

### CODE REVIEW

File: docs/ui/layout-containers.md [hunk 1]
```diff
@@ -484,3 +563,143 @@ None.
...
+| alignContent | Gets or sets a value that helps aligning a flex container's lines within it when there is extra space in the cross-axis, similar to how justifyContent aligns individual items within the main-axis. The alignContent property accepts 6 different values: `flex-start` (lines packed to the start of the container), `flex-end` (lines packed to the end of the container), `center` (lines packed to the center of the container), `space-between` (lines evenly distributed; the first line is at the start of the container while the last one is at the end), `space-around` (lines evenly distributed with equal space between them), and `stretch` (lines stretch to take up the remaining space). The default value is `stretch`. This property has no effect when the flexbox has only a single line. |
+
+> **NOTE:** There is a limitation for `alignItems` in **iOS**, the `baseline` option can **not** be used.
```
Comment (tjvantoll) [inline docs/ui/layout-containers.md]: There is a limitation for `alignItems` in **iOS**—the `baseline` option can **not** be used.

Technically using a comma is incorrect here, although it’s not really a big deal.

File: docs/ui/layout-containers.md [hunk 2]
```diff
@@ -484,3 +563,143 @@ None.
...
+| alignSelf | Gets or sets a value that makes it possible to override the alignItems value for specific flex items. This property accepts the same 5 values as the alignItems: `flex-start` (cross-start margin edge of the item is placed on the cross-start line), `flex-end` (cross-end margin edge of the item is placed on the cross-end line), `center` (item is centered in the cross-axis), `baseline` (items are aligned such as their baseline are aligned), and `stretch` (stretch to fill the container but still respect min-width / max-width). The default value is `stretch`. |
+| flexWrapBefore | Gets or sets a boolean value controlling item wrapping. Setting it to `true` on flexbox item will force it to wrap on a new line. The default value is `false`. This property is not part of the official flexbox specification.  
+
+> **NOTE:** There is a limitation for `alignSelf` in **iOS**, the `baseline` option can **not** be used.
```
Comment (tjvantoll) [inline docs/ui/layout-containers.md]: Same note as before, switch the comma to an em dash or colon.

**Review by tjvantoll — APPROVED** [review]
This is excellent @manoldonev. We can merge this after the Angular snippets are merged in.

This doesn’t relate directly to this PR, but I do find it a bit weird that we store a bunch of these snippets in the unit tests of external repositories, as it makes it harder for people to contribute.

**Comment by manoldonev** [comment]
@tjvantoll removed the commas.
I agree it makes it harder to contribute -- in fact I first created the JSON files with snippets in a local folder but then noticed the local folder is git-ignored so I remembered @vchimev mentioning once that such snippets are kept in the test apps [so it can be ensured snippets at least do not throw after modifications to the codebase]. So I browsed the build scripts to get an idea where exactly should I place them.

**Comment by vchimev** [comment]
docs

**Comment by vchimev** [comment]
docs

**Comment by vchimev** [comment]
```
Liquid Exception: Snippet file not found: /home/nsbuilduser/workspace/docs-stable-PR/build/Content/snippets/flexbox-layout-flexdirection-row.json in ui/layout-containers.md
jekyll 3.0.1 | Error:  Snippet file not found: /home/nsbuilduser/workspace/docs-stable-PR/build/Content/snippets/flexbox-layout-flexdirection-row.json
```

**Comment by manoldonev** [comment]
@vchimev @etabakov https://github.com/NativeScript/nativescript-angular/pull/1124 has to go first as the flexbox snippets are new.

**Comment by manoldonev** [comment]
docs

**Comment by manoldonev** [comment]
@vchimev @tjvantoll https://github.com/NativeScript/nativescript-angular/pull/1124 is merged, docs checks passed so I am merging this.

**Pull request merged**